sycl/doc/extensions/SPIRV/SPV_INTEL_arbitrary_precision_floating_point.asciidoc

```diff
@@ -0,0 +1,1160 @@
+Note that the exponent values (E1, Eout) are inferred from the width of the `OpTypeInt`.
+
+_EnableSubnormals_ specifies whether subnormal numbers should be supported or flushed to zero before and after the operation and is chosen from _Table 3.16a_.
+
+_RoundingMode_ controls the rounding mode for the result and is chosen from _Table 3.16_.
+
+_RoundingAccuracy_ controls the accuracy of the rounding operation and is chosen from _Table 3.16d_.
+
+| Capability:
+*ArbitraryPrecisionFloatingPointINTEL*
+| 9 | 5841 | <id> Result Type | Result <id> | A <id> | _Literal_ M1 | _Literal_ Mout | _Literal_ EnableSubnormals | _Literal_ RoundingMode | _Literal_ RoundingAccuracy
+|=====
+
+[cols="9", width="100%"]
+|=====
+8+<|*OpArbitraryFloatCastFromIntINTEL* +
+
+An `OpTypeInt` representing an arbitrary precision integer is passed in as _A_.
+It is type casted into an arbitrary precision floating point number with the new specification (Eout, Mout) and returned as _Result_.
+
+_Result Type_ is an `OpTypeInt` of width `1+Eout+Mout` and is the type of _Result_.
+
+_Eout_ and _Mout_ contain the width of the exponent and the mantissa of the floating point type of _Result_.
+Note that the exponent value (Eout) is inferred from the width of the `OpTypeInt`.
+
+_EnableSubnormals_ specifies whether subnormal numbers should be supported or flushed to zero before and after the operation and is chosen from _Table 3.16a_.
+
+_RoundingMode_ controls the rounding mode for the result and is chosen from _Table 3.16_.
+
+_RoundingAccuracy_ controls the accuracy of the rounding operation and is chosen from _Table 3.16d_.
+
+| Capability:
+*ArbitraryPrecisionFloatingPointINTEL*
+| 8 | 5842 | <id> Result Type | Result <id> | A <id> | _Literal_ Mout | _Literal_ EnableSubnormals | _Literal_ RoundingMode | _Literal_ RoundingAccuracy
+|=====
+
+[cols="9", width="100%"]
+|=====
+8+<|*OpArbitraryFloatCastToIntINTEL* +
+
+An `OpTypeInt` representing an arbitrary precision floating point number (ihc_float) is passed in as _A_.
+It is type casted into an arbitrary precision integer with width `W` and returned as _Result_.
+
+_Result Type_ is an `OpTypeInt` of width `W` and is the type of _Result_.
+
+_E1_ and _M1_ contain the width of the exponent and the mantissa of the floating point type of _A_.
+Note that the exponent value (E1) is inferred from the width of the `OpTypeInt`.
+
+_EnableSubnormals_ specifies whether subnormal numbers should be supported or flushed to zero before and after the operation and is chosen from _Table 3.16a_.
+
+_RoundingMode_ controls the rounding mode for the result and is chosen from _Table 3.16_.
+
+_RoundingAccuracy_ controls the accuracy of the rounding operation and is chosen from _Table 3.16d_.
+
+| Capability:
+*ArbitraryPrecisionFloatingPointINTEL*
+| 8 | 5843 | <id> Result Type | Result <id> | A <id> | _Literal_ M1 | _Literal_ EnableSubnormals | _Literal_ RoundingMode | _Literal_ RoundingAccuracy
+|=====
+
+=== Validation Rules
+
+None.
```

Comment:
Are you sure that there are no validation rules?

I think I can imagine at least a few:
- `M*` literal arguments can't contains values that exceed width of `OpTypeInt` of corresponding argument minus 1
- `M*` literal should contain value more than zero

It seems that there are no requirement to have arguments of the same width, but probably there are still some limitations related to this?

Note: the same applies to fixed-point extension

Reply:
Good point - added for floating point. Based on the discussion about what values `I` can take, I don't know if this applies to the fixed_point case. What do you think?